libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/PemKeyConfig.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.X509ExtendedKeyManager;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A {@link SslKeyConfig} that reads from PEM formatted paths.
+ */
+public final class PemKeyConfig implements SslKeyConfig {
+    private final Path certificate;
+    private final Path key;
+    private final char[] keyPassword;
+
+    public PemKeyConfig(Path certificate, Path key, char[] keyPassword) {
+        this.certificate = Objects.requireNonNull(certificate, "Certificate cannot be null");
+        this.key = Objects.requireNonNull(key, "Key cannot be null");
+        this.keyPassword = Objects.requireNonNull(keyPassword, "Key password cannot be null (but may be empty)");
+    }
+
+    @Override
+    public Collection<Path> getDependentFiles() {
+        return Arrays.asList(certificate, key);
+    }
+
+    @Override
+    public X509ExtendedKeyManager createKeyManager() {
+        try {
+            PrivateKey privateKey = getPrivateKey();
+            List<Certificate> certificates = getCertificates();
+            final KeyStore keyStore = KeyStoreUtil.buildKeyStore(certificates, privateKey, keyPassword);
+            return KeyStoreUtil.createKeyManager(keyStore, keyPassword, KeyManagerFactory.getDefaultAlgorithm());
+        } catch (RuntimeException e) {
```

Comment:
I think we should move `getPrivateKey` and `getCertificates` out of the try catch block since they appear to handle the exceptions. Then I think this code would only need to handle the `GeneralSecurityException`?

Reply:
> Why do we catch and rethrow a RuntimeException

So that the catch block for `Exception` only catches checked exceptions.
I'll update it to be more explicit.